File: game_server/ServerProtocol/src/Common/ChildEnum/EAdultStatus.java
package Common.ChildEnum;

/*********
 * 子嗣状态
 **/
public enum EAdultStatus {
	NONE, //0 ==== 
	APPLY_PLAYER, //1 ==== 指定联姻请求中
	APPLY_SERVER, //2 ==== 全服联姻请求中
	MARRIED, //3 ==== 已婚
	;
public static final EAdultStatus[]  EAdultStatus_Values = EAdultStatus.values();
public static final int EAdultStatus_Length = EAdultStatus_Values.length;
public static EAdultStatus EAdultStatus_FromInt(int _ivalue) {
	if(_ivalue < 0 || _ivalue >= EAdultStatus_Length){ return null; }
	return EAdultStatus_Values[_ivalue];
}
}

File: game_server/ClientProtocol/Common/ChildEnum/EAdultStatus.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using ALBasicProtocolPack;
using UnityEngine;


namespace Common.ChildEnum
{

/// <summary>
/// 子嗣状态
/// </summary>
public enum EAdultStatus {
	NONE, //0 ==== 
	[InspectorName("APPLY_PLAYER - idx[1] - 指定联姻请求中")]
	APPLY_PLAYER, //1 ==== 指定联姻请求中
	[InspectorName("APPLY_SERVER - idx[2] - 全服联姻请求中")]
	APPLY_SERVER, //2 ==== 全服联姻请求中
	[InspectorName("MARRIED - idx[3] - 已婚")]
	MARRIED, //3 ==== 已婚
}

public class EAdultStatusComparer : IEqualityComparer<EAdultStatus>{
	public bool Equals(EAdultStatus x, EAdultStatus y) { return x == y; }
	public int GetHashCode(EAdultStatus obj) { return obj.GetHashCode(); }
	public static int g_iEnumCount = 4;
}
}

